nautobot/extras/tests/test_views.py

```diff
@@ -988,28 +1036,234 @@ def setUpTestData(cls):
+    def test_run_later_missing_name(self, patched):
+        self.add_permissions("extras.run_job")
+
+        data = {
+            "_schedule_type": "future",
+        }
+
+        patched.return_value = 1
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}), data)
+
+        self.assertHttpStatus(response, 200)
+        errors = extract_form_failures(response.content.decode(response.charset))
+        self.assertEqual(errors, ["_schedule_name: Please provide a name for the job schedule."])
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    @mock.patch("nautobot.extras.utils.get_worker_count")
+    def test_run_later_missing_date(self, patched):
+        self.add_permissions("extras.run_job")
+
+        data = {
+            "_schedule_type": "future",
+            "_schedule_name": "test",
+        }
+
+        patched.return_value = 1
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}), data)
+
+        self.assertHttpStatus(response, 200)
+        errors = extract_form_failures(response.content.decode(response.charset))
+        self.assertEqual(
+            errors,
+            [
+                "_schedule_start_time: Please enter a valid date and time greater than or equal to the current date and time."
+            ],
+        )
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    @mock.patch("nautobot.extras.utils.get_worker_count")
+    def test_run_later_date_passed(self, patched):
+        self.add_permissions("extras.run_job")
+
+        data = {
+            "_schedule_type": "future",
+            "_schedule_name": "test",
+            "_schedule_start_time": str(datetime.now() - timedelta(minutes=1)),
+        }
+
+        patched.return_value = 1
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}), data)
+
+        self.assertHttpStatus(response, 200)
+        errors = extract_form_failures(response.content.decode(response.charset))
+        self.assertEqual(
+            errors,
+            [
+                "_schedule_start_time: Please enter a valid date and time greater than or equal to the current date and time."
+            ],
+        )
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    @mock.patch("nautobot.extras.utils.get_worker_count")
+    def test_run_later(self, patched):
+        self.add_permissions("extras.run_job")
+
+        start_time = timezone.now() + timedelta(minutes=1)
+        data = {
+            "_schedule_type": "future",
+            "_schedule_name": "test",
+            "_schedule_start_time": str(start_time),
+        }
+
+        patched.return_value = 1
+        response = self.client.post(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}), data)
+
+        self.assertHttpStatus(response, 302)
```

Comment:
assertRedirects - this will be a different redirect than the run_immediately tests right?